client/components/slotMaker.jsx

```diff
@@ -1,41 +1,58 @@
 import React from 'react';
 
 const time = [
-  '5:00 am', '5:30 am', '6:00 am', '6:30 am', '7:00 am',
-  '7:30 am', '8:00 am', '8:30 am', '9:00 am', '9:30 am',
-  '10:00 am', '10:30 am', '11:00 am', '11:30 am', '12:00pm',
-  '1:00 pm', '1:30 pm', '2:00 pm', '2:30 pm', '3:00 pm',
-  '3:30 pm', '4:00 pm', '4:30 pm','5:00 pm', '5:30 pm',
-  '6:00 pm', '6:30 pm', '7:00 pm', '7:30 pm', '8:00 pm',
-  '8:30 pm', '9:00 pm', '9:30 pm', '10:00 pm', '10:30 pm',
-  '11:00 pm'
+  '5:00 AM', '5:30 AM', '6:00 AM', '6:30 AM', '7:00 AM',
+  '7:30 AM', '8:00 AM', '8:30 AM', '9:00 AM', '9:30 AM',
+  '10:00 AM', '10:30 AM', '11:00 AM', '11:30 AM', '12:00PM',
+  '1:00 PM', '1:30 PM', '2:00 PM', '2:30 PM', '3:00 PM',
+  '3:30 PM', '4:00 PM', '4:30 PM','5:00 PM', '5:30 PM',
+  '6:00 PM', '6:30 PM', '7:00 PM', '7:30 PM', '8:00 PM',
+  '8:30 PM', '9:00 PM', '9:30 PM', '10:00 PM', '10:30 PM',
+  '11:00 PM'
 ];
-let SlotMaker = (props) => {
-  let timeAvail = [];
-  let notAvail = [];
-  let timeObjArray = props.timeSlots;
-  let idxOfPick = time.findIndex(pick => pick === props.time);
-  let j = idxOfPick;
-  if (props.clicked === true) {
-    for (var i = 0; i < timeObjArray.length; i++) {
-      notAvail.push(timeObjArray[i].time);
+class SlotMaker extends React.Component { 
+  constructor (props) {
+    super(props);
+    this.state = {
+      timeAvail: [],
     }
-    for (var i = idxOfPick; i < idxOfPick + 5; i++) {
-      if (!notAvail.includes(time[i]) && time[i] !== undefined) {
-        timeAvail.push(time[i]);
+  }
+  render () {
+    let timeAvail = [];
+    let notAvail = [];
+    let timeObjArray = this.props.timeSlots;
+    let idxOfPick = time.findIndex(pick => pick === this.props.time);
+    let j = idxOfPick + 1;
+    if (this.props.partySizeMax <= this.props.partySize) {
+      return (
+        <div className="jumbotron"></div>
+      )
+    }
+    if (this.props.clicked === true && this.state.counterClick !== this.props.timeSlots) {
+      for (var i = 0; i < timeObjArray.length; i++) {
+        notAvail.push(timeObjArray[i].time);
       }
-      j--;
-      if (!notAvail.includes(time[j]) && time[j] !== undefined) {
-        timeAvail.unshift(time[j]);
+      for (var i = idxOfPick; i < idxOfPick + 6; i++) {
+        if (!notAvail.includes(time[i]) && time[i] !== undefined) {
+          timeAvail.push(time[i]);
+        }
+        j--;
+        if (!notAvail.includes(time[j]) && time[j] !== undefined) {
+          timeAvail.unshift(time[j]);
+        }
       }
+      this.setState({
```

Comment:
Not really sure what's going on in this render function, but I don't think you should be calling "this.setState()" inside  a render function of your component because that will trigger a re-render again even though it hasn't finished rendering the first time. May be wrong about that tho